Pod/Classes/NSURLSession/Matchers/SimpleRequestMatcher.swift

```diff
@@ -9,32 +9,47 @@
 import Foundation
 
 /**
- This struct matches by matching the path to a regular expression and by 
+ This struct matches by matching the path to a regular expression and by
  comparing the methods for equality.
-*/
+ */
 struct SimpleRequestMatcher : RequestMatcher {
-    
+
     let pathMatcher: NSRegularExpression
-    
+
     let method: String
-    
+
     init(url: NSURL, method: String) {
         let path = NSRegularExpression.escapedPatternForString(url.absoluteString)
         try! self.init(expression: "^\(path)$", method: method)
     }
-    
+
     init(expression: String, method: String) throws {
         try self.pathMatcher = NSRegularExpression(pattern: expression, options: NSRegularExpressionOptions.AnchorsMatchLines)
         self.method = method
     }
-    
-    func matches(request: NSURLRequest) -> Bool {
-        guard request.HTTPMethod == self.method else { return false }
-        
+
+    func matches(request: NSURLRequest) -> MatchesResponse {
+        //
+        guard request.HTTPMethod == self.method else { return .NoMatch }
+
+        // Get the match
         let path = request.URL?.absoluteString ?? ""
-        let options = NSMatchingOptions(rawValue: 0)
         let range = NSMakeRange(0, path.utf16.count)
-        return pathMatcher.numberOfMatchesInString(path, options: options, range: range) == 1
+        let matches = self.pathMatcher.matchesInString(path, options: [], range: range)
+        guard let match = matches.first where matches.count == 1 else { return .NoMatch }
```

Comment:
Does this mean that if I set a pattern like `/product/` and the url is `/product/123/product/` the return will be `.NoMatch`?